### PR TITLE
Speedup static collection during development

### PR DESCRIPTION
**Pull request by snoepkast**
The speedup is achieved by only doing the collection once per request instead of once for every time the template tag is used. For reference see issue #428.

### CODE REVIEW

File: pipeline/collector.py [hunk 1]
```diff
@@ -24,21 +26,23 @@ def clear(self, path=""):
...
-                # Prefix the relative path if the source storage contains it
-                if getattr(storage, 'prefix', None):
-                    prefixed_path = os.path.join(storage.prefix, path)
-                else:
-                    prefixed_path = path
-                if prefixed_path not in found_files:
-                    found_files[prefixed_path] = (storage, path)
-                    self.copy_file(path, prefixed_path, storage)
+    def collect(self, request=None):
+        if not self.request or not self.request is request:  # Prevent execution multiple times per request
```
Comment (cyberdelia) [inline pipeline/collector.py]: Could this be a guard?

```python
if self.request and self.request is request:
    return
self.request = request
found_files = OrdereDict()
[...]
```

**Comment by cyberdelia** [comment]
Looks good, thanks a lot :bowtie: 
It would be lovely if the tests passed :neutral_face:

**Comment by snoepkast** [comment]
I don't get why my change would break this? Do you have any idea?

**Comment by cyberdelia** [comment]
@snoepkast My current guess, is that your changes in ``PipelineMixin`` have broke ``PipelineExtension``, probably because you changed ``__init__``.

**Comment by leonardoo** [comment]
@snoepkast the problem is the __init__ like say @cyberdelia, you can convert init get request_var to something like this

```python
@property
    def request_var(self):
        if not self._request_var:
            self._request_var = template.Variable('request')
        return self._request_var
```

and remove the init call in these methods

```python
class StylesheetNode(PipelineMixin, template.Node):
    def __init__(self, name):
         super(StylesheetNode, self).__init__(name) # this call is the problem
         self.name = name
  class JavascriptNode(PipelineMixin, template.Node):
     def __init__(self, name):
         super(JavascriptNode, self).__init__(name) # this call is the problem
         self.name = name

```

**Comment by snoepkast** [comment]
Locally the tests run now, but apparently not on Travis.. But it fails on a native python 'open file' which has nothing to do with any of this?

**Comment by snoepkast** [comment]
Ahh now I see the PR fixing that problem (#459), so this code should work then.

**Comment by cyberdelia** [comment]
Merged in 840bf27af1aa690c49b39499f7f6980f408e455e

**Comment by Remiz** [comment]
Amazing work! Now everything is loading under a second for me.

Thank you so much for this PR guys!

**Pull request closed without merging**